plugins/security/nexus-ssl-plugin/src/main/java/com/sonatype/nexus/ssl/plugin/internal/ui/TrustStoreComponent.groovy

```diff
@@ -75,9 +101,65 @@ extends DirectComponentSupport
    */
   @DirectMethod
   @RequiresAuthentication
+  @RequiresPermissions('nexus:ssl:truststore:delete')
   @Validate
   void remove(final @NotEmpty(message = '[id] may not be empty') String id) {
-    trustStoreResource.delete(id)
+    try {
+      trustStore.removeTrustCertificate(id)
+    }
+    catch (KeyNotFoundException e) {
+      throw new Exception("Certificate with fingerprint '${id}' not found")
+    }
+  }
+
+  static CertificateXO asCertificateXO(final Certificate certificate, final boolean inNexusSSLTrustStore)
```

Comment:
I did tested. The method is used also by CertificateResource, but can be package scoped